server/routes/api.js

```diff
@@ -23,13 +24,55 @@ module.exports = (services) => {
       throw err;
     }
   });
-
+  /* POST calculate */
   router.post('/calculate', (req, res) => {
     const dijkstra = new Dijkstra(req.body.from, req.body.dest, { MAX_DISTANCE: 4 });
     const result = dijkstra.solve();
     console.log(result);
     res.json(result);
   });
+  /* POST routes */
+  router.post('/routes', async (req, res) => {
+    try {
+      const routes = req.body;
+      const trip = {
+        tracknum: uuidv4(),
+        status: 'inprogress',
+      };
+
+      // store trip
+      const tripResult = await services.db.trip.create(trip);
+      const tripid = tripResult.id;
+
+      for (let i = 0, len = routes.length; i < len; i += 1) {
+        if (i !== len - 1) {
+          const segment = {
+            source_id: routes[i].id,
+            des_id: routes[i + 1].id,
+            trip_id: tripid,
+            drone_id: null,
+          };
+          // store segments
+          await services.db.segment.create(segment);
+        }
+
+        if (routes[i].name === 'source' || routes[i].name === 'destination') {
+          const route = {
+            type: routes[i].name,
+            latitude: routes[i].lat,
+            longitude: routes[i].lng,
+          };
+          // store places
+          await services.db.place.create(route);
+        }
+      }
+
+      res.status(200).send('Successful!');
+    } catch (err) {
+      throw err;
```

Comment:
You shouldn't immediately throw an error you just caught. This makes the catch block pointless. Instead return an error response so the caller of this endpoint knows the call failed.

Reply:
@bill-feng Thanks for pointing it out, didn't know!:) Can you explain what exactly I should do here? `'Instead return an error response so the caller of this endpoint knows the call failed'`

Reply:
@bill-feng Thanks for the comment, didn't know!:) Returned an stat code and error message in response.